flowtracker/src/main/scala/FlowTracker.scala

```diff
@@ -144,11 +145,12 @@ class FlowTracker(val flow: Flow[_],
           The job will run as normal, but it will not be tracked by Sahale.
           """.stripMargin.trim, t)
         runCompleted.set(true)
+        isTrackable = false
     }
 
     val maxFailures = 10
     var numFailures = 0
-    while(!runCompleted.get) {
+    while(!runCompleted.get && isTrackable) {
```

Comment:
The `isTrackable` check isn't strictly required here, but included it for consistency.